pyramid_rpc/jsonrpc.py

```diff
@@ -160,6 +160,44 @@ def __call__(self, context, request):
         return result
 
 
+class jsonrpc_batch_tween(object):
+    """ Pyramid Tween used to intercept requests and handle batch requests
+    transparently for pyramid_rpc in accordance with the JSONRPC 2.0 Specification.
+
+    """
+    def __init__(self, handler, registry):
+        self.handler = handler
+        self.registry = registry
+
+    def __call__(self, request):
+        # Check if it is a JSON request and a list.
+        try:
+            json_body = request.json_body[:]
+        except (ValueError, TypeError):
+            return self.handler(request)
+
+        response_body = []
+        for json_request in json_body:
+            # request.json_body is only a decorator applying json.loads to request.body
+            request.body = json.dumps(json_request)
+
+            # Keep the last response object and extract the json
+            # from the body.
+            response = self.handler(request)
+            response_body.append(json.loads(response.body))
+
+        response_body = [result for result in response_body if result != '']
+        if len(response_body) == 0:
+            # If no response objects are available the JSONRPC Specification forbids
+            # sending an empty array. An empty string is sent instead.
+            response.body = '""'
```

Comment:
[If there are no Response objects contained within the Response array as it is to be sent to the client, the server MUST NOT return an empty Array and should return nothing at all.](http://www.jsonrpc.org/specification#batch)